src/library/commissioner_impl.cpp

```diff
@@ -1161,10 +1183,17 @@ Error CommissionerImpl::SignRequest(coap::Request &aRequest, tlv::Scope aScope)
 
     VerifyOrExit(IsCcmMode(), error = Error::kInvalidState);
 
+#if OT_COMM_CCM_ENABLE
     SuccessOrExit(error = mTokenManager.SignMessage(signature, aRequest));
 
     SuccessOrExit(error = AppendTlv(aRequest, {tlv::Type::kCommissionerToken, mTokenManager.GetToken(), aScope}));
     SuccessOrExit(error = AppendTlv(aRequest, {tlv::Type::kCommissionerSignature, signature, aScope}));
+#else
+    (void)aRequest;
```

Comment:
unreachable code? should have `goto exit` by `VerifyOrExit(IsCcmMode(), error = Error::kInvalidState);`

Reply:
`IsCcmMode` stands for `the Commissioner is configured to run in CCM mode`, I think we should reject a CCM configuration when `OT_COMM_CCM` is set to `OFF`.

Reply:
yes, we should reject `CCM` configuration so that when `OT_COMM_CCM=0`, `IsCcmMode()` always returns `FALSE`. 
Thoughts?